### DISCUSSION TITLE
How to generate a new test case?

### DISCUSSION THREAD

**Discussion by EdwardWong1994** [j178/leetgo#220] [General]
Hello, i have two questions waiting for any kind help.

1. If i want to abitary test some case in local machine, not just the test cases that leetcode provide us. Is there a way to do so??

2. it seems that i can not use `gcc` to output `solution.o` and that means i can not debug in my local machine.
Is there a way to sort these two problems out??

Much thanks

**Comment by j178** [comment]
1. Just add your test case to `testcases.txt`, ensuring to include `input:` and `output:` markers like this:
   ```
   input:
   param_a
   param_b
   output:
   output_a
   ```

2. I don't quite understand this question. The generated `solution.cpp` is a fully functional program that can be built and debugged using any compiler or debugger of your choice.

**Reply by EdwardWong1994** [reply]
Thanks for your instant reply.  For the first question, i did add test case in `testcases.txt` and run the `solution.cpp` file,  but it will not consider my added cases.  Still the default cases leetcode provide.

The content in the`txt`file is like this:

```

input:
3
output:
[[1,2,3],[8,9,4],[7,6,5]]

input:
1
output:
[[1]]


input:
2
output:
[[1,2],[4,3]]

The last case is what i add in the `txt` file, the previous two were default.  But when i enter `leetcode test 59`, the following screenshot shows that the added cases were not tested.

```


<img width="699" alt="image" src="https://github.com/j178/leetgo/assets/96438179/b05fe7ec-a189-4490-b9d7-a91785c6dfad">


For the second questions, I use `g++ -std=C+17 example.cpp -o example.o` , it shows error in the following screenshot.

<img width="776" alt="image" src="https://github.com/j178/leetgo/assets/96438179/45b603cd-27e8-4f25-8e17-de23a111c10e">

can i just comment this libaray file?  I do not know whether it have influences?

**Reply by j178** [reply]
1. Currently, `testcases.txt` is only used for local testing (`leetgo test -L`). By default (`leetgo test`), tests are run remotely on LeetCode judge server, which is why the cases you added are not working. I will fix this issue in the next version.
2. You might consider upgrading your gcc to v8.3 or a later version. However, you can safely comment out the include line as a easier  solution.

**Comment by EdwardWong1994** [comment]
> 1. Currently, `testcases.txt` is only used for local testing (`leetgo test -L`). By default (`leetgo test`), tests are run remotely on LeetCode judge server, which is why the cases you added are not working. I will fix this issue in the next version.
> 2. You might consider upgrading your gcc to v8.3 or a later version. However, you can safely comment out the include line as a easier  solution.

yes, nicely solve the problem
